pkg/virtctl/imageupload/imageupload.go

```diff
@@ -177,6 +179,12 @@ func parseArgs(args []string) error {
 		return fmt.Errorf("cannot upload to a readonly volume, use either ReadWriteOnce or ReadWriteMany if supported")
 	}
 
+	var err error
+	volumeMode, err = normalizeVolumeMode(volumeMode, blockVolume)
```

Comment:
Can you pass in `cmd.Flags()` from where `parseArgs` is called?

You can then simply do this:

```golang
if cmd.Flags().Lookup("block-volume").Changed && cmd.Flags().Lookup("volume-mode").Changed {
    return fmt.Errorf("'--block-volume' and '--volume-mode' are mutually exclusive")
} else if cmd.Flags().Lookup("block-volume").Changed {
 // use "block-volume"
} else {
// use "volume-mode"; either the supplied value or default will be use
}
```

Reply:
Made the changes according to your suggestion.